### PR TITLE
[Diagnostics] Extend fix-it logic for misssing arguments without parenthesis

### PR DESCRIPTION
**Pull request by janbaig**
When diagnosing missing arguments without parentheses, the fix-it insertion location can be ambiguous. This update improves the logic by inserting necessary parentheses around the missing argument for the fix-it. If an insertion location cannot be determined, the function now returns early to prevent incorrect suggestions, enhancing error handling precision.

The test suite has been updated to reflect these changes.

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test Windows platform

**Comment by xedin** [comment]
@swift-ci please test Linux platform

File: lib/Sema/CSDiagnostics.cpp [hunk 1]
```diff
@@ -5338,6 +5338,8 @@ bool MissingArgumentsFailure::diagnoseSingleMissingArgument() const {
     // fn(argX, argY):
     //   fn(argX, argY[, argMissing])
     if (args->empty()) {
+      if (!args->getRParenLoc().isValid())
+        return false;
```
Comment (xedin) [inline lib/Sema/CSDiagnostics.cpp:5342]: I think this is too drastic. If there are no parens we can still produce a diagnostic and suggest a fix-it. The diagnostic can be attached to the end of the "fn" - `fnExpr->getEndLoc()` and suggest adding parens.

**Review by xedin — CHANGES_REQUESTED** [review]
I left a comment inline but we also need a test-case that we can use to make sure that fix-it works correctly for "no arguments" case.

**Comment by JanBaig** [comment]
@swift-ci please test

**Review by xedin — APPROVED** [review]

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test Windows platform

**Pull request merged**